R/get_types_recipes.R

```diff
@@ -0,0 +1,112 @@
+## Buckets variables into discrete, mutally exclusive types
+get_types <- function(x) {
```

Comment:
First major change is happening here. Get types now we return multiple types for each variables instead of just picking one. This means that all the function around recipes that deal with types now are made to work lists instead of character vectors

``` r
# CRAN
library(recipes)
tmp <- tibble::tibble(
  ord = as.ordered(letters),
  chr = letters,
  int = seq_along(letters),
  dbl = seq_along(letters) / 5
)

class(tmp$ord)
#> [1] "ordered" "factor"
#> [1] "ordered" "factor"

recipes:::get_types(tmp)
#> # A tibble: 4 × 2
#>   variable type   
#>   <chr>    <chr>  
#> 1 ord      nominal
#> 2 chr      nominal
#> 3 int      numeric
#> 4 dbl      numeric
```

``` r
# PR
library(recipes)

tmp <- tibble::tibble(
  ord = as.ordered(letters),
  chr = letters,
  int = seq_along(letters),
  dbl = seq_along(letters) / 5
)

class(tmp$ord)
#> [1] "ordered" "factor"

recipes:::get_types(tmp)
#> # A tibble: 4 × 2
#>   variable type     
#>   <chr>    <list>   
#> 1 ord      <chr [2]>
#> 2 chr      <chr [3]>
#> 3 int      <chr [2]>
#> 4 dbl      <chr [2]>

recipes:::get_types(tmp)$type
#> [[1]]
#> [1] "ordered" "nominal"
#> 
#> [[2]]
#> [1] "string"    "unordered" "nominal"  
#> 
#> [[3]]
#> [1] "integer" "numeric"
#> 
#> [[4]]
#> [1] "double"  "numeric"
```